hack/build-and-push.sh

```diff
@@ -431,6 +434,34 @@ attach_migration_file() {
     return 0
 }
 
+# Find previous bundle that has a migration and output its digest.
+find_previous_migration_bundle_digest() {
+    local -r task_name=${1:?Missing task name}
+    local -r task_version=${2:?Missing task version}
+    local -r repo=${TEST_REPO_NAME:-task-${task_name}}
+
+    local -r url_list_recent_tags="https://quay.io/api/v1/repository/${QUAY_NAMESPACE}/${repo}/tag/?onlyActiveTags=true&limit=100"
```

Comment:
Why do we need to get a 100 of bundles? If the latest one has the annotation, we know the bundle with the most recent migration from it's value.

Reply:
The reason to list multiple tags is for the testing where task and pipeline bundles are pushed into a single image repository. How the tasks and pipelines are mixed is unpredictable. 100 is too big actually. For convenience, how about reduce it into a smaller one, like 30 or 20? Hard to tell the new number can cover all possible cases, but it could for most of time I think.

Reply:
As discussed today with the team, let's assume that each task has own image repository.

Reply:
Reduced it to 30.

Note: the limit may be needed to be increased when `build-and-push.sh` runs for a single task without setting `BUILD_TAG`.

Reply:
@mmorhun PTAL